blocks/form/form.js

```diff
@@ -0,0 +1,217 @@
+function createSelect(fd) {
+  const select = document.createElement('select');
+  select.id = fd.Field;
+  if (fd.Placeholder) {
+    const ph = document.createElement('option');
+    ph.textContent = fd.Placeholder;
+    ph.setAttribute('selected', '');
+    ph.setAttribute('disabled', '');
+    select.append(ph);
+  }
+  fd.Options.split(',').forEach((o) => {
+    const option = document.createElement('option');
+    option.textContent = o.trim();
+    option.value = o.trim();
+    select.append(option);
+  });
+  if (fd.Mandatory === 'x') {
+    select.setAttribute('required', 'required');
+  }
+  return select;
+}
+
+function constructPayload(form) {
+  const payload = {};
+  [...form.elements].forEach((fe) => {
+    if (fe.type === 'checkbox') {
+      if (fe.checked) payload[fe.id] = fe.value;
+    } else if (fe.id) {
+      payload[fe.id] = fe.value;
+    }
+  });
+  return payload;
+}
+
+async function submitForm(form) {
+  const payload = constructPayload(form);
+
+  const resp = await fetch(form.dataset.action, {
+    method: 'POST',
+    cache: 'no-cache',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ data: payload }),
+  });
+  await resp.text();
+  return payload;
+}
+
+function checkValidity(form) {
+  const payload = constructPayload(form);
+  const { email } = payload;
+  const regex = /^[^@]+@(?!(yahoo|hotmail|gmail|icloud|outlook))[^@]+\.[a-z]{2,}$/;
+  return String(email).match(regex);
+}
+
+function createButton(fd) {
+  const button = document.createElement('button');
+  button.textContent = fd.Label;
+  button.classList.add('button');
+  button.classList.add('primary');
+  if (fd.Type === 'submit') {
+    button.addEventListener('click', async (event) => {
+      const form = button.closest('form');
+      if (fd.Placeholder) form.dataset.action = fd.Placeholder;
+      if (form.checkValidity() && checkValidity(form) !== null) {
+        event.preventDefault();
```

Comment:
Looks like this is duplicated in both sides of the `if`.

Reply:
Done.

Reply:
I still see `event.preventDefault();` duplicated in the `if` and the `else`. It can be moved above to avoid duplication.

Reply:
Done.